llama_index/llms/vllm.py

```diff
@@ -0,0 +1,362 @@
+import json
+from typing import Any, Dict, List, Optional, Sequence
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.callbacks import CallbackManager
+from llama_index.llms.base import (
+    LLM,
+    ChatMessage,
+    ChatResponse,
+    ChatResponseAsyncGen,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseAsyncGen,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.llms.vllm_utils import get_response, post_http_request
+
+
+class Vllm(LLM):
+    model: Optional[str] = Field(description="The HuggingFace Model to use.")
+
+    temperature: float = Field(description="The temperature to use for sampling.")
+
+    tensor_parallel_size: Optional[int] = Field(
+        default=1,
+        description="The number of GPUs to use for distributed execution with tensor parallelism.",
+    )
+
+    trust_remote_code: Optional[bool] = Field(
+        default=True,
+        description="Trust remote code (e.g., from HuggingFace) when downloading the model and tokenizer.",
+    )
+
+    max_retries: int = Field(default=10, description="The maximum number of retries.")
+
+    n: int = Field(
```

Comment:
maybe a better name like num_output_seq

Reply:
actually that option is redundant i will remove it .

Reply:
Actually that parameter is redundant I will remove it .